### PR TITLE
Institution configuration options in aggregate

### PR DESCRIPTION
**Pull request by arothuis**

### CODE REVIEW

File: src/Surfnet/Stepup/Tests/Configuration/InstitutionConfigurationTest.php [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+use Surfnet\Stepup\Configuration\Event\NewInstitutionConfigurationCreatedEvent;
+use Surfnet\Stepup\Configuration\Event\ShowRaaContactInformationOptionChangedEvent;
+use Surfnet\Stepup\Configuration\Event\UseRaLocationsOptionChangedEvent;
+use Surfnet\Stepup\Configuration\InstitutionConfiguration;
+use Surfnet\Stepup\Configuration\Value\Institution;
+use Surfnet\Stepup\Configuration\Value\InstitutionConfigurationId;
+use Surfnet\Stepup\Configuration\Value\ShowRaaContactInformationOption;
+use Surfnet\Stepup\Configuration\Value\UseRaLocationsOption;
+
+class InstitutionConfigurationTest extends AggregateRootScenarioTestCase
```
Comment (arothuis) [inline src/Surfnet/Stepup/Tests/Configuration/InstitutionConfigurationTest.php]: These tests can move to a CommandHandler test once there is one.

**Comment by arothuis** [comment]
@DRvanR 👀

File: src/Surfnet/Stepup/Configuration/Api/InstitutionConfiguration.php [hunk 2]
```diff
@@ -25,15 +25,36 @@
...
+     * @param UseRaLocationsOption $useRaLocationsOption
+     * @return void
+     */
+    public function changeUseRaLocationsOption(UseRaLocationsOption $useRaLocationsOption);
+
+    /**
+     * @param ShowRaaContactInformationOption $raaContactInformationOption
+     * @return void
+     */
+    public function changeShowRaaContactInformationOption(ShowRaaContactInformationOption $raaContactInformationOption);
```
Comment (DRvanR) [inline src/Surfnet/Stepup/Configuration/Api/InstitutionConfiguration.php]: `set` vs `change`? Seems change would imply a change, regardless of whether or not there actually is a change... set is a bit of a nono, however, you're setting an option, something that is actually quite normal to do. Anyways, might want to give it some thought. Not saying it has to change, just that there might be a better alternative...

File: src/Surfnet/Stepup/Configuration/InstitutionConfiguration.php [hunk 3]
```diff
@@ -58,15 +62,44 @@ class InstitutionConfiguration extends EventSourcedAggregateRoot implements Inst
...
      */
-    public static function create(InstitutionConfigurationId $institutionConfigurationId, Institution $institution)
-    {
+    public static function create(
+        InstitutionConfigurationId $institutionConfigurationId,
+        Institution $institution,
+        UseRaLocationsOption $useRaLocationsOption = null,
+        ShowRaaContactInformationOption $showRaaContactInformationOption = null
+    ) {
+        if ($useRaLocationsOption === null) {
```
Comment (DRvanR) [inline src/Surfnet/Stepup/Configuration/InstitutionConfiguration.php]: why the nullable? Is there a way to create this *with* options?

**Comment by DRvanR** [comment]
two lil' things.

**Comment by arothuis** [comment]
@DRvanR, I removed the optional arguments.

You described my thought process on the `set` vs `change` exactly. In parlance, I would prefer setting an option. However, that could be confused with setters in code (which they explicitly are not). I also considered `toggle` but that feels more like the general process of inverting a binary option. For lack of a better alternative, I would keep it this way.

**Comment by DRvanR** [comment]
:+1:

**Pull request merged**